### DISCUSSION TITLE
umi开启微前端之后，主应用调试子应用的代码，如何开启source-map？

### DISCUSSION THREAD

**Discussion by senfish** [umijs/umi#11287] [Q&A]
"umi": "3.5.20"
    "@umijs/plugin-qiankun": "2.39.2",
    
    主应用的.umirc.ts
    {
         // mfsu: {},
         qiankun: {
                slave: {},
         },
        devtool: process.env.NODE_ENV === 'development' ? 'eval-cheap-module-source-map' : 'source-map',
    }
   子应用的devtool配置跟主应用一样。
   但是调试代码的时候，还是显示是压缩之后的代码。
   1. 我尝试开启mfsu或者关闭mfsu，并且删除了.umi重跑之后也没有什么效果
   2. 我都给devtool设置成'source-map'，并且删除了.umi重跑之后，还是没什么效果

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4
